webapp/src/Controller/API/ProblemController.php

```diff
@@ -95,9 +95,11 @@ public function listAction(Request $request): Response
 
             $objects = [];
             foreach ($ordinalArray->getItems() as $item) {
-                /** @var ContestProblemWrapper $contestProblemWrapper */
-                $contestProblemWrapper = $item->getItem();
-                $contestProblem        = $contestProblemWrapper->getContestProblem();
+                /** @var ContestProblemWrapper|ContestProblem $contestProblem */
+                $contestProblem = $item->getItem();
+                if ($contestProblem instanceof ContestProblemWrapper) {
+                    $contestProblem = $contestProblem->getContestProblem();
```

Comment:
So we don't test jury calls (yet) I assume?

Reply:
Is indeed a good further improvement. Would need to see if that’s already possible with the BaseTest to test for multiple users?

Reply:
Ah probably not yet, but can be an addition later on

Reply:
It's not fully trivial. You could do something like this:

make $apiUser an array which can maximally contain [null, demo, admin]. When unset defaults to [null]. Give expectedObjects and expectedAbsent an extra dimension which specifies for each user the expected objects or absent objects.

Within the BaseTest tests, loop over the users in the array and repeat each test for each of the users and compare it to the corresponding specified objects.

Reply:
I've devised a different solution for now.